src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Recipes/ElasticIndexRebuildStep.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.BackgroundJobs;
+using OrchardCore.Recipes.Models;
+using OrchardCore.Recipes.Services;
+using OrchardCore.Search.Elasticsearch.Core.Services;
+
+namespace OrchardCore.Search.Elasticsearch.Core.Recipes
+{
+    /// <summary>
+    /// This recipe step rebuilds an Elasticsearch index.
+    /// </summary>
+    public class ElasticIndexRebuildStep : IRecipeStepHandler
+    {
+        public async Task ExecuteAsync(RecipeExecutionContext context)
+        {
+            if (!String.Equals(context.Name, "elastic-index-rebuild", StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            var indices = context.Step["Indices"];
+
+            if (indices != null)
+            {
+                await HttpBackgroundJob.ExecuteAfterEndOfRequestAsync("elastic-index-rebuild", async scope => {
```

Comment:
@Skrypt  This will not work I'm afraid. (I had same problem in one of my steps)

Inside **ExecuteAfterEndOfRequestAsync** method there is a shell state check which will prevent this method to execute.

Before recipe executor starts its job, we are changing shell state to **Initializing** so all subsequent requests respond with 503 (service unavailable).

![image](https://user-images.githubusercontent.com/6403130/189493920-f81abbb9-71fa-441e-88ca-c2f59a5a3c1e.png)

Here is [my solution](https://github.com/OrchardCMS/OrchardCore/pull/12302/files#diff-698892fd391652965297a199f6de6d899264f9ecb8464e8ad6155795b8aba68f) to this problem, I've added optional parameter to **ExecuteAfterEndOfRequestAsync** which allow us to bypass this state check.

Reply:
I think I removed the state change into the controller that was causing the issue on my side after discussing with @jtkech